pkg/argocd/git.go

```diff
@@ -44,7 +51,16 @@ func TemplateCommitMessage(tpl *template.Template, appName string, changeList []
 	// writer of a template.
 	changes := make([]commitMessageChange, 0)
 	for _, c := range changeList {
-		changes = append(changes, commitMessageChange{c.Image.ImageName, c.OldTag.String(), c.NewTag.String()})
+		newTag := c.NewTag.String()
+		image := c.Image.ImageName
+		applicationName, CommitId := GetApplicationDetails(newTag, image)
```

Comment:
this shouldn't be the responsibility of the template function. This was a pure function.